apis/Google.Datastore.V1Beta3/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs

```diff
@@ -90,10 +91,11 @@ public void StructuredQuery()
             {
                 Filter = Filter.Equal("author", "Jane Austen")
             };
-            DatastoreQueryResults results = db.RunQuery(query);
-            // DatastoreQueryResults implements IEnumerable<Entity>, but you can
-            // call AsEntityResults(), AsBatches() or AsResponses() to see the query
-            // results in whatever way makes most sense for your application.
+            LazyDatastoreQuery results = db.RunQuery(query);
```

Comment:
I imagine it's quite frustrating to hear me offer alternative after alternative on this point, but I think I finally found one where the users assumptions match the actual RPCs generated.

db.RunQuery(query) returns all the results full fetched, like db.RunQuery.GetAllResults() does currently.
db.IterateQuery(query) returns an IEnumerator<T> (not an IEnumerable<T>).

Then, my assumptions match the underlying implementation.  Specifically:
1. RunQuery() fetches all the results now.
2. IterateQuery() starts fetching results and will fetch more as I iterate over the results.
3. I will never unknowingly initiate an RPC by iterating over results.